android/app/src/main/java/dev/fredag/cheerwithme/data/FriendsRepository.kt

```diff
@@ -3,15 +3,31 @@ package dev.fredag.cheerwithme.data
 import dev.fredag.cheerwithme.data.backend.*
 import kotlinx.coroutines.flow.Flow
 import kotlinx.coroutines.flow.flow
-import kotlinx.coroutines.flow.onStart
 import javax.inject.Inject
 
+sealed class Result<T> {
+    class Ok<T>(val r: T) : Result<T>()
+    class Err<T>(val e: String) : Result<T>()
+}
+
 class FriendsRepository @Inject constructor(
     private val backendService: BackendService
 ) {
-    fun getFriends(): Flow<UserFriends> = flow {
-        val friends = backendService.friends()
-        emit(friends)
+    fun getFriends(): Flow<Result<UserFriends>> = flow {
+        try {
+            val request = backendService.friends()
```

Comment:
Could we move the body of this try to another method? quite a lot of indentation here